addon-resizer/nanny/kubernetes_client.go

```diff
@@ -85,28 +89,35 @@ func (k *kubernetesClient) countNodesThroughMetrics() (uint64, error) {
 		return 0, err
 	}
 
-	decoder := expfmt.SampleDecoder{
-		Dec:  expfmt.NewDecoder(strings.NewReader(string(rawMetrics)), expfmt.FmtText),
-		Opts: &expfmt.DecodeOptions{},
-	}
-	var v model.Vector
+	decoder := expfmt.NewDecoder(strings.NewReader(string(rawMetrics)), expfmt.FmtText)
```

Comment:
It should be 
```
decoder := expfmt.NewDecoder(bytes.NewReader(rawMetrics), expfmt.FmtText)
```

Calling string(byteBuffer) allocates memory, I wrote this simple benchmark to confirm that - https://github.com/mm4tt/sandbox/blob/master/gobench/bench_test.go#L40 
So calling string here doubles the memory we need to store rawMetrics.

Reply:
Thanks - though no longer valid after my next change. PTAL